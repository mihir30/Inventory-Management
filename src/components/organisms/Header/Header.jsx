import React from 'react';
import MenuRoundedIcon from '@mui/icons-material/MenuRounded';
import './Header.scss';
const Header = (props) => {
    const manage = () => {
        props.sethamstate(!props.hamstate);
    }
    return <header className='header' aria-label='header'>
        <div className='header-box'>
            <div className='left-aligned'>
                <div className='hamburger' onClick={manage}>
                    <MenuRoundedIcon fontSize='medium' />
                </div>
                <div className='app-title' data-testid="title">
                    INVENTORY APP
                </div>
            </div>
            <div className='owner-box'>
                <span className='owner-logo'>O</span>
                <div className='app-owner' data-testid="owner">
                    Owner
                </div>
            </div>
        </div>
    </header>
}
export default Header;