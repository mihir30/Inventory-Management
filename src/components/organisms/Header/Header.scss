.header {
    // width: 100%;
    display: flex;
}
.header-box {
    margin: 0 auto;
    display: flex;
    background-color: white;
    color: #475466;
    border: 0.1px solid #dadee0;
    justify-content: space-between;
    align-items: center;
    width: 100%;
    flex-direction: row;
}
.btn {
    margin: 0.2rem;
}
.hamburger {
    cursor: pointer;
    margin-left: 1rem;
    padding: 0.2rem;
}
.hamburger:hover {
    background-color: lightgrey;
    border-radius: 100%;
    transform: scale(0.9);
    transition: 1ms;
}
.left-aligned {
    display: flex;
    justify-content: space-between;
    align-items: center;
}
.hamburger {
    display: flex;
    justify-content: space-between;
}
.app-title {
    display: flex;
    font-weight: 500;
    font-size: medium;
    align-items: flex-start;
    margin: 0 1.5rem;
}
.owner-box {
    display: flex;
    justify-content: space-between;
    align-items: center;
}
.app-owner {
    font-weight: 500;
    font-size: medium;
    margin-right: 2rem;
    cursor: pointer;
}
.owner-logo {
    background-color: #1ca1c1;
    border-radius: 100px;
    width: 43px;
    height: 43px;
    color: #ffffff;
    text-align: center;
    line-height: 44px;
    font-size: 18px;
    margin: 0.5rem 1rem;
    cursor: pointer;
}

