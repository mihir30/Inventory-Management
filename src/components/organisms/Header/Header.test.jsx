import Header from "./Header";
import ReactDom from "react-dom";
import { render, screen } from '@testing-library/react';



describe("Header Component", () => {

    it("should render the component properly", () => {
        const headerEl = document.createElement("header");
        ReactDom.render(<Header/>, headerEl); 
    });

    it('should have the right title', () => {
        render(<Header/>);
        expect(screen.getByTestId("title")).toHaveTextContent("INVENTORY APP");
    })
    it('should have the right text', () => {
        render(<Header/>);
        expect(screen.getByTestId("owner")).toHaveTextContent("Owner");
    })
});