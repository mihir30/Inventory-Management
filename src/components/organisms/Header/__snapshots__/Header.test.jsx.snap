exports[`Header Component should have the right header component snapshot 1`] = `
<header
  aria-label="header"
  className="header"
>
<div className='header-box'>
</div>
</header>